### DISCUSSION TITLE
Metasploit automation

### DISCUSSION THREAD

**Discussion by Jackson-Pollock** [rapid7/metasploit-framework#15346] [Q&A] [answered]
Is it possible to integrate Metasploit framework with custom hunting platforms? Where attacker will choose exploits and options from that GUI and that platform will pass those inputs to Metasploit in backend and Metasploit execute attacks and hunting platform will show the output accordingly? 
Hunting platform in front-end and Metasploit in backend.

**Comment by bcoles** [comment, marked as answer]
Check out MSF RPC:

* https://docs.rapid7.com/metasploit/rpc-api/
* https://docs.rapid7.com/metasploit/sample-usage-of-the-rpc-api/

Alternatively, `msfconsole` is a command line application (and supports resource scripts), so you could call it from the command line. But I wouldn't recommend it. RPC is a better approach.
